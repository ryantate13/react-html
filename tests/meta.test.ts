import {meta} from '../src';

describe('meta', () => {

    it('renders with empty attributes', () => {
        expect(meta({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(meta()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(meta({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(meta({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(meta({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(meta({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(meta({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(meta({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(meta({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(meta({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(meta({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(meta({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(meta({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(meta({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(meta({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(meta({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(meta({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(meta({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(meta({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(meta({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(meta({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(meta({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(meta({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(meta({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(meta({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(meta({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(meta({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(meta({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(meta({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(meta({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(meta({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(meta({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(meta({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(meta({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(meta({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(meta({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(meta({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(meta({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(meta({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(meta({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(meta({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(meta({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(meta({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(meta({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(meta({translate: "no"})).toMatchSnapshot();
    });

    it('renders with charSet="utf-8"', () => {
        expect(meta({charSet: "utf-8"})).toMatchSnapshot();
    });

    it('renders with content="TEST"', () => {
        expect(meta({content: "TEST"})).toMatchSnapshot();
    });

    it('renders with httpEquiv="content-type"', () => {
        expect(meta({httpEquiv: "content-type"})).toMatchSnapshot();
    });

    it('renders with httpEquiv="default-style"', () => {
        expect(meta({httpEquiv: "default-style"})).toMatchSnapshot();
    });

    it('renders with httpEquiv="refresh"', () => {
        expect(meta({httpEquiv: "refresh"})).toMatchSnapshot();
    });

    it('renders with httpEquiv="x-ua-compatible"', () => {
        expect(meta({httpEquiv: "x-ua-compatible"})).toMatchSnapshot();
    });

    it('renders with httpEquiv="content-security-policy"', () => {
        expect(meta({httpEquiv: "content-security-policy"})).toMatchSnapshot();
    });

    it('renders with name="TEST"', () => {
        expect(meta({name: "TEST"})).toMatchSnapshot();
    });
});