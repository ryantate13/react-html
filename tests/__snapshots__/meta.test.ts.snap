// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`meta renders with accessKey="a,b,c,d" 1`] = `
<meta
  accessKey="a,b,c,d"
/>
`;

exports[`meta renders with autoCapitalize="characters" 1`] = `
<meta
  autoCapitalize="characters"
/>
`;

exports[`meta renders with autoCapitalize="none" 1`] = `
<meta
  autoCapitalize="none"
/>
`;

exports[`meta renders with autoCapitalize="off" 1`] = `
<meta
  autoCapitalize="off"
/>
`;

exports[`meta renders with autoCapitalize="on" 1`] = `
<meta
  autoCapitalize="on"
/>
`;

exports[`meta renders with autoCapitalize="sentences" 1`] = `
<meta
  autoCapitalize="sentences"
/>
`;

exports[`meta renders with autoCapitalize="words" 1`] = `
<meta
  autoCapitalize="words"
/>
`;

exports[`meta renders with charSet="utf-8" 1`] = `
<meta
  charSet="utf-8"
/>
`;

exports[`meta renders with className="foo bar baz" 1`] = `
<meta
  className="foo bar baz"
/>
`;

exports[`meta renders with content="TEST" 1`] = `
<meta
  content="TEST"
/>
`;

exports[`meta renders with contentEditable="false" 1`] = `
<meta
  contentEditable="false"
/>
`;

exports[`meta renders with contentEditable="true" 1`] = `
<meta
  contentEditable="true"
/>
`;

exports[`meta renders with dir="auto" 1`] = `
<meta
  dir="auto"
/>
`;

exports[`meta renders with dir="ltr" 1`] = `
<meta
  dir="ltr"
/>
`;

exports[`meta renders with dir="rtl" 1`] = `
<meta
  dir="rtl"
/>
`;

exports[`meta renders with draggable="false" 1`] = `
<meta
  draggable="false"
/>
`;

exports[`meta renders with draggable="true" 1`] = `
<meta
  draggable="true"
/>
`;

exports[`meta renders with empty attributes 1`] = `<meta />`;

exports[`meta renders with hidden=false 1`] = `
<meta
  hidden={false}
/>
`;

exports[`meta renders with hidden=true 1`] = `
<meta
  hidden={true}
/>
`;

exports[`meta renders with httpEquiv="content-security-policy" 1`] = `
<meta
  httpEquiv="content-security-policy"
/>
`;

exports[`meta renders with httpEquiv="content-type" 1`] = `
<meta
  httpEquiv="content-type"
/>
`;

exports[`meta renders with httpEquiv="default-style" 1`] = `
<meta
  httpEquiv="default-style"
/>
`;

exports[`meta renders with httpEquiv="refresh" 1`] = `
<meta
  httpEquiv="refresh"
/>
`;

exports[`meta renders with httpEquiv="x-ua-compatible" 1`] = `
<meta
  httpEquiv="x-ua-compatible"
/>
`;

exports[`meta renders with id="TEST" 1`] = `
<meta
  id="TEST"
/>
`;

exports[`meta renders with inputMode="decimal" 1`] = `
<meta
  inputMode="decimal"
/>
`;

exports[`meta renders with inputMode="email" 1`] = `
<meta
  inputMode="email"
/>
`;

exports[`meta renders with inputMode="none" 1`] = `
<meta
  inputMode="none"
/>
`;

exports[`meta renders with inputMode="numeric" 1`] = `
<meta
  inputMode="numeric"
/>
`;

exports[`meta renders with inputMode="search" 1`] = `
<meta
  inputMode="search"
/>
`;

exports[`meta renders with inputMode="tel" 1`] = `
<meta
  inputMode="tel"
/>
`;

exports[`meta renders with inputMode="text" 1`] = `
<meta
  inputMode="text"
/>
`;

exports[`meta renders with inputMode="url" 1`] = `
<meta
  inputMode="url"
/>
`;

exports[`meta renders with is="foo" 1`] = `
<meta
  is="foo"
/>
`;

exports[`meta renders with itemID="https://google.com/" 1`] = `
<meta
  itemID="https://google.com/"
/>
`;

exports[`meta renders with itemProp="https://google.com/" 1`] = `
<meta
  itemProp="https://google.com/"
/>
`;

exports[`meta renders with itemRef="foo bar" 1`] = `
<meta
  itemRef="foo bar"
/>
`;

exports[`meta renders with itemScope=false 1`] = `
<meta
  itemScope={false}
/>
`;

exports[`meta renders with itemScope=true 1`] = `
<meta
  itemScope={true}
/>
`;

exports[`meta renders with itemType="https://google.com/" 1`] = `
<meta
  itemType="https://google.com/"
/>
`;

exports[`meta renders with lang="" 1`] = `
<meta
  lang=""
/>
`;

exports[`meta renders with lang="es-419" 1`] = `
<meta
  lang="es-419"
/>
`;

exports[`meta renders with name="TEST" 1`] = `
<meta
  name="TEST"
/>
`;

exports[`meta renders with slot="TEST" 1`] = `
<meta
  slot="TEST"
/>
`;

exports[`meta renders with spellCheck="false" 1`] = `
<meta
  spellCheck="false"
/>
`;

exports[`meta renders with spellCheck="true" 1`] = `
<meta
  spellCheck="true"
/>
`;

exports[`meta renders with style={"color":"#000000"} 1`] = `
<meta
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`meta renders with tabIndex=1 1`] = `
<meta
  tabIndex={1}
/>
`;

exports[`meta renders with title="TEST" 1`] = `
<meta
  title="TEST"
/>
`;

exports[`meta renders with translate="no" 1`] = `
<meta
  translate="no"
/>
`;

exports[`meta renders with translate="yes" 1`] = `
<meta
  translate="yes"
/>
`;

exports[`meta renders without attributes 1`] = `<meta />`;
